doc/api/async_context.md

```diff
@@ -0,0 +1,607 @@
+# Asynchronous Context Tracking
+
+> Stability: 2 - Stable
+
+<!-- source_link=lib/async_hooks.js -->
+
+## Introduction
+These classes are used to create asynchronous state within callbacks and promise
+chains. They allow storing data throughout the lifetime of a web request
+or any other asynchronous duration. It is similar to thread-local storage
+in other languages.
+
+The `AsyncLocalStorage` and `AsyncResource` classes are part of the
+`async_hooks` module:
+
+```js
+const async_hooks = require('async_hooks');
+```
+
+## Class: `AsyncLocalStorage`
+<!-- YAML
+added:
+ - v13.10.0
+ - v12.17.0
+changes:
+ - version: REPLACEME
+   pr-url: https://github.com/nodejs/node/pull/37675
+   description: AsyncLocalStorage is now Stable. Previously, it had been Experimental.
+-->
+
+This class is used to create asynchronous state within callbacks and promise
+chains. It allows storing data throughout the lifetime of a web request
+or any other asynchronous duration. It is similar to thread-local storage
+in other languages.
```

Comment:
This is basically a copy/paste of the introduction paragraph, so maybe should reword one or the other so as to not be so redundant.